lib/src/usage.dart

```diff
@@ -133,118 +130,103 @@ class Usage {
     return result;
   }
 
-  String getAllowedTitle(Option option, String allowed) {
+  String _allowedTitle(Option option, String allowed) {
     var isDefault = option.defaultsTo is List
         ? option.defaultsTo.contains(allowed)
         : option.defaultsTo == allowed;
     return '      [$allowed]' + (isDefault ? ' (default)' : '');
   }
 
-  void calculateColumnWidths() {
+  List<int> _calculateLineWidths() {
     var abbr = 0;
     var title = 0;
-    for (var option in optionsAndSeparators) {
+    for (var option in _optionsAndSeparators) {
       if (option is! Option) continue;
       if (option.hide) continue;
 
       // Make room in the first column if there are abbreviations.
-      abbr = math.max(abbr, getAbbreviation(option).length);
+      abbr = math.max(abbr, _abbreviation(option).length);
 
       // Make room for the option.
-      title = math.max(title, getLongOption(option).length);
+      title = math.max(title, _longOption(option).length);
 
       // Make room for the allowed help.
       if (option.allowedHelp != null) {
         for (var allowed in option.allowedHelp!.keys) {
-          title = math.max(title, getAllowedTitle(option, allowed).length);
+          title = math.max(title, _allowedTitle(option, allowed).length);
         }
       }
     }
 
     // Leave a gutter between the columns.
     title += 4;
-    columnWidths = [abbr, title];
+    return [abbr, title];
   }
 
-  void newline() {
-    newlinesNeeded++;
-    currentColumn = 0;
-    numHelpLines = 0;
+  void _newline() {
+    _newlinesNeeded++;
+    _currentColumn = 0;
   }
 
-  void write(int column, String text) {
+  void _write(int column, String text) {
     var lines = text.split('\n');
     // If we are writing the last column, word wrap it to fit.
-    if (column == columnWidths.length && lineLength != null) {
-      var wrappedLines = <String>[];
-      var start = columnWidths
-          .sublist(0, column)
-          .reduce((start, width) => start += width);
-
-      for (var line in lines) {
-        wrappedLines
-            .addAll(wrapTextAsLines(line, start: start, length: lineLength));
-      }
-
-      lines = wrappedLines;
+    if (column == _columnWidths.length && lineLength != null) {
+      var start =
+          _columnWidths.take(column).reduce((start, width) => start + width);
+      lines = [
+        for (var line in lines)
+          ...wrapTextAsLines(line, start: start, length: lineLength),
+      ];
     }
 
     // Strip leading and trailing empty lines.
-    while (lines.isNotEmpty && lines[0].trim() == '') {
-      lines.removeRange(0, 1);
```

Comment:
Haha, wow, This is *old* Dart code. (Dart used to not have `removeAt()` because someone on the team was morally opposed to allowing users the possibility of accidentally calling that in a loop instead of the more efficient `removeRange()`.)